addons/interaction/functions/fnc_passMagazine.sqf

```diff
@@ -0,0 +1,49 @@
+/*
+ * Author: BaerMitUmlaut
+ * Pass spare magazine for the specified weapon.
+ *
+ * Arguments:
+ * 0: Unit that passes the magazine <OBJECT>
+ * 1: Unit to pass the magazine to <OBJECT>
+ * 2: Weapon classname <STRING>
+ * 
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_player, _target, "arifle_MX_F"] call ace_interaction_fnc_magToPassazine
+ *
+ * Public: No
+ */
+
+#include "script_component.hpp"
+params ["_player", "_target", "_weapon"];
+private ["_compatibleMags", "_filterFunc", "_filteredMags", "_magToPass", "_magToPassIndex"];
+
+_compatibleMags = getArray (configfile >> "CfgWeapons" >> _weapon >> "magazines");
+_filterFunc = compile format ["((_this select 0) in %1) && !(_this select 2)", _compatibleMags];
+
+_filteredMags = [magazinesAmmoFull _player, _filterFunc] call EFUNC(common,filter);
```

Comment:
I assumed I couldn't use the local variables, my bad.

You not only need to check if the player has a compatible magazine for the other unit, you also need to check if he passes one of those compatible magazines. Say an ammo carrier carries both 30rnd 5.56 mags and also 20rnd 7.62 mags for a DMR. If I'd skip that check, he'd always pass the DMR a 5.56 magazine because it has more rounds.